pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go

```diff
@@ -246,12 +256,380 @@ func TestDeploymentService_executeK8sSyncStage_withInputNamespace(t *testing.T)
+	require.NoError(t, err)
+
+	require.Equal(t, "piped", service.GetLabels()["pipecd.dev/managed-by"])
+	require.Equal(t, "piped-id", service.GetLabels()["pipecd.dev/piped"])
+	require.Equal(t, "app-id", service.GetLabels()["pipecd.dev/application"])
+	require.Equal(t, "0123456789", service.GetLabels()["pipecd.dev/commit-hash"])
+
+	require.Equal(t, "simple", service.GetName())
+	require.Equal(t, "piped", service.GetAnnotations()["pipecd.dev/managed-by"])
+	require.Equal(t, "piped-id", service.GetAnnotations()["pipecd.dev/piped"])
+	require.Equal(t, "app-id", service.GetAnnotations()["pipecd.dev/application"])
+	require.Equal(t, "v1", service.GetAnnotations()["pipecd.dev/original-api-version"])
+	require.Equal(t, "0123456789", service.GetAnnotations()["pipecd.dev/commit-hash"])
+
+	target := filepath.Join("./", "testdata", "prune_with_change_namespace", "target")
+
+	// read the running application config from the example file
+	targetCfg, err := os.ReadFile(filepath.Join(target, "app.pipecd.yaml"))
+	require.NoError(t, err)
+
+	// prepare the request to ensure the running deployment exists
+	targetRequest := &deployment.ExecuteStageRequest{
+		Input: &deployment.ExecutePluginInput{
+			Deployment: &model.Deployment{
+				PipedId:       "piped-id",
+				ApplicationId: "app-id",
+				DeployTargets: []string{"default"},
+			},
+			Stage: &model.PipelineStage{
+				Id:   "stage-id",
+				Name: "K8S_SYNC",
+			},
+			StageConfig: []byte(``),
+			RunningDeploymentSource: &deployment.DeploymentSource{
+				ApplicationDirectory:      running,
+				CommitHash:                "0123456789",
+				ApplicationConfig:         runningCfg,
+				ApplicationConfigFilename: "app.pipecd.yaml",
+			},
+			TargetDeploymentSource: &deployment.DeploymentSource{
+				ApplicationDirectory:      target,
+				CommitHash:                "0012345678",
+				ApplicationConfig:         targetCfg,
+				ApplicationConfigFilename: "app.pipecd.yaml",
+			},
+		},
+	}
+
+	resp, err = svc.ExecuteStage(ctx, targetRequest)
+	require.NoError(t, err)
+	require.Equal(t, model.StageStatus_STAGE_SUCCESS.String(), resp.GetStatus().String())
+
+	// The service should be removed from the previous namespace
+	_, err = dynamicClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}).Namespace("test-1").Get(context.Background(), "simple", metav1.GetOptions{})
+	require.Error(t, err)
+
+	// The service should be created in the new namespace
+	service, err = dynamicClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}).Namespace("test-2").Get(context.Background(), "simple", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	require.Equal(t, "piped", service.GetLabels()["pipecd.dev/managed-by"])
+	require.Equal(t, "piped-id", service.GetLabels()["pipecd.dev/piped"])
+	require.Equal(t, "app-id", service.GetLabels()["pipecd.dev/application"])
+	require.Equal(t, "0012345678", service.GetLabels()["pipecd.dev/commit-hash"])
```

Comment:
[imo] In addition to them, we need to check the resource key to check whether the namespace has changed correcty.